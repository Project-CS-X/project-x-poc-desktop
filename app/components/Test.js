import React from 'react';

function Test(props) {
    return (
        <div>
            <h2>Test Component 1</h2>
        </div>
    )
}

export default Test