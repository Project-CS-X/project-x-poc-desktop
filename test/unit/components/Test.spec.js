/* eslint react/jsx-props-no-spreading: off */
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import Test from '../../../app/components/Test'

Enzyme.configure({ adapter: new Adapter() })

function setup() {
 
  const component = shallow(<Test />)

  return {
    component,
    header:  component.find('h2')
  }
}

describe('Test component', () => {
  it('should have a header that says \'Test Component 1\'', () => {
    const { header } = setup()
    expect(header.text()).toMatch(/^Test Component 1$/)
  })

  // Snap
  it('should match exact snapshot', () => {
    const test = (
        <Test />
    )

    const tree = renderer.create(test).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
