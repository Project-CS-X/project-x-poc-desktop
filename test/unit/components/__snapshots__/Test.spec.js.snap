// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component should match exact snapshot 1`] = `
<div>
  <h2>
    Test Component
  </h2>
</div>
`;
